examples/bri-3/src/bri/vsm/capabilites/handleWorkstepFailures/handleWorkstepFailuresCommand.handler.ts

```diff
@@ -0,0 +1,45 @@
+import { CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+import { MessagingAgent } from '../../../communication/agents/messaging.agent';
+import { BpiMessage } from '../../../communication/models/bpiMessage';
+import { BpiMessageType } from '../../../communication/models/bpiMessageType.enum';
+import { HandleWorkstepFailuresCommand } from './handleWorkstepFailures.command';
+import { TransactionStorageAgent } from '../../../transactions/agents/transactionStorage.agent';
+
+@CommandHandler(HandleWorkstepFailuresCommand)
+export class HandleWorkstepFailuresCommandHandler
+  implements ICommandHandler<HandleWorkstepFailuresCommand>
+{
+  constructor(
+    private readonly messagingAgent: MessagingAgent,
+    private txStorageAgent: TransactionStorageAgent,
+  ) {}
+
+  async execute(command: HandleWorkstepFailuresCommand) {
+    const errPayload = {
+      errorCode: 'xxx',
+      errorMessage: 'Execution fails',
+    };
+
+    const tx = await this.txStorageAgent.getTransactionById(command.id);
+
+    if (tx != undefined) {
+      const errorBpiMessage = new BpiMessage(
+        tx.id,
+        tx.fromBpiSubjectAccountId,
+        tx.toBpiSubjectAccountId,
+        JSON.stringify(errPayload),
+        tx.signature,
+        BpiMessageType.Err,
+      );
+
+      const initiatorChannel =
+        tx.fromBpiSubjectAccount.ownerBpiSubject.publicKey;
+      await this.messagingAgent.publishMessage(
+        initiatorChannel,
+        JSON.stringify(errorBpiMessage),
+      );
+      tx.updateStatusToAborted();
```

Comment:
We want to keep this and the line bellow in the original command handler, to make sure it is handled under a single db transaction later. I guess we do not need the HandleWorkstepFailuresCommand and handler at all and can instead log the issue, constrict the bpi message and publish it to the initiator channel in the WorkstepExecutionFailuresEventHandler

Reply:
@ognjenkurtic I have made adjustments according to your comments. Right now, I think saga is not necessary in this PR. Would you like me to remove that part as well?

Reply:
Thanks, looks great. left a couple of smaller comments and it is ready to merge AFAIC. You can remote the saga.

Reply:
@ognjenkurtic Done. Please review it. Thank you!